docs/style-spec/expressions.md

```diff
@@ -0,0 +1,148 @@
+{
+  "circle-color": {
+    "expression": [
+      'rgb',
+      // red is higher when feature.properties.temperature is higher
+      ["number_data", "temperature"],
+      0,
+      // blue is higher when feature.properties.temperature is lower
+      ["-", 100, ["number_data", "temperature"]]
+    ]
+  }
+}
+```
+
+
+## Zoom-dependent functions
+
+A <strong>zoom function</strong> is any function defined using an expression that includes a reference to `["zoom"]`.  Such functions allow the appearance of a map feature change with map’s zoom level. Zoom functions can be used to create the illusion of depth and control data density.  
+
+A zoom function must be of the following form (see [Curves](#Curves) below):
+
+```js
+{
+  "expression": [ "curve", interpolation, ["zoom"], ... ]
+}
+```
+
+(`["zoom"]` may not appear anywhere else in the expression.)
+
+
+### Example: a zoom-only function.
+
+```js
+{
+  "circle-radius": {
+    "expression": [
+      "curve", "linear", ["zoom"],
+      // zoom is 5 (or less) -> circle radius will be 1px
+      // zoom is 10 (or greater) -> circle radius will be 2px
+      5, 1, 10, 2
+    ]
+  }
+}
+```
+
+### Example: a zoom-and-property function
+
+Using property functions as the output value for one or more zoom stops allows 
+the appearance of a map feature to change with both the zoom level _and_ the
+feature's properties.
+
+```js
+{
+  "circle-radius": {
+    "expression": [
+      "curve", "linear", ["zoom"],
+
+      // zoom is 0 and "rating" is 0 -> circle radius will be 0px
+      // zoom is 0 and "rating" is 5 -> circle radius will be 5px
+      0, [ "number_data", "rating" ],
+      // zoom is 20 and "rating" is 0 -> circle radius will be 4 * 0 = 0px
+      // zoom is 20 and "rating" is 5 -> circle radius will be 4 * 5 = 20px
+      10, [ "*", 4, ["number_data", "rating"] ]
+    ]
+  }
+}
+```
+
+## Property Expressions
+
+Property expressions are represented using a Lisp-like structured syntax tree.
+
+**Constants:**
+- `[ "ln2" ]`
+- `[ "pi" ]`
+- `[ "e" ]`
+
+**Literals:**
+- string `"foo"`
+- number `42`
+- boolean `true` or `false`
+- null `null`
+- object `["object", obj]` where `obj` is a JSON object literal
+- array `["array", arr]` where `arr` is a JSON array literal
+
+**Lookup:**
+- `["get", container, key ]`
+  - `container`: `array` or `object`
+  - `key`: `string` if `container` is an object, `number` if it's an array.
+- `[ "has", container, key ]` returns `true` if the property is present, false otherwise.
+- `[ "typeof", container, key ]` yields the data type of `container[key]`: one of `'string'`, `'number'`, `'boolean'`, `'object'`, `'array'`, or, in the case that the property is not present, `'none'`.
+
+**Feature data:**
+- `["properties"]` the feature's `properties` object
+- `["geometry_type"]` the string value of `feature.geometry.type`
+- `[ "id" ]` returns the value of `feature.id`.
+
+**Decision:**
+- `["case", cond1, result1, cond2, result2, ..., ["else"], result_m]`
```

Comment:
It's not necessary, but I personally think it (a) doesn't do much harm, (b) makes parsing slightly easier, (c) makes mistakes easier to spot.  Guessing I'll be outvoted on this though.